### PR TITLE
User story 30 pd

### LINKED ISSUE #32: User Story 30, User cancels an order
As a registered user
When I visit an order's show page
I see a button or link to cancel the order only if the order is still pending
When I click the cancel button for an order, the following happens:
- Each row in the "order items" table is given a status of "unfulfilled"
- The order itself is given a status of "cancelled"
- Any item quantities in the order that were previously fulfilled have their quantities returned to their respective merchant's inventory for that item.
- I am returned to my profile page
- I see a flash message telling me the order is now cancelled
- And I see that this order now has an updated status of "cancelled"

### PR DESCRIPTION
**Pull request by philjdelong**
## Description
Resolves #32 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Refactor (non-breaking change which maintains existing functionality)
  (order_controller, method: .update)
- [x] New feature (no-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## RSpec Results
**ALL UNCOVERED CODE IS EITHER:
     - SAD PATH
     - Part 2 - STORIES**
```
Finished in 5.75 seconds (files took 2.36 seconds to load)
161 examples, 0 failures

Coverage report generated for RSpec to /Users/phildelong/turing/mod2/projects/monster_shop_part_1/coverage. 2
029 / 2036 LOC (99.66%) covered.
```

### CODE REVIEW

**Review by mel-rob — APPROVED** [review]
Looks great, thanks!

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -36,10 +36,20 @@ def create
...
+        redirect_to "/profile"
+      elsif current_user.admin?
+        redirect_to "/admin"
+      elsif current_user.merchant?
+        redirect_to "/merchant"
+      elsif current_user.default?
+        redirect_to "/profile"
+      else
+        flash[:error] = order.errors.full_messages.to_sentence
+      end
```
Comment (DanielEFrampton) [inline app/controllers/orders_controller.rb:51]: That works! Let's refactor it later in the project to pull some of this stuff into helper methods.

File: app/models/item_order.rb [hunk 2]
```diff
@@ -15,4 +17,10 @@ def change_order_to_packaged
       order.packaged!
     end
   end
+
+  def change_to_unfulfilled
+    if self.fulfilled?
+      item.update(inventory: (item.inventory + quantity))
+    end
```
Comment (DanielEFrampton) [inline app/models/item_order.rb:24]: I'm a little confused about how the order of these things works, but the test looks solid so I guess it's working! (and I was there when we designed this, ha!)

**Review by DanielEFrampton — APPROVED** [review]
Thanks for tackling this one, it was really confusing. Solid test.

**Pull request merged**